DS-5171 As a SM I want to keep Revisions for Data Policy

OH17L opened the pull request.

ohorbatiuk committed:
DS-5171 by chmez: Rename entity type for storing user consents.

ohorbatiuk committed:
DS-5171 by chmez: Create entity type for storing versions of data policy description.

ohorbatiuk committed:
DS-5171 by chmez:

1. Delete old configuration form for edit description of data policy.

2. Hide data policy entities overview page in menu.

3. Allow opening revisions page of data policy entity directly from the administrative menu.

4. Create a special form for edit data policy description and add a link to this form on revisions overview page.

5. Show list of user consents on a page of related data policy revision.

ohorbatiuk committed:
Merge branch 'develop' of github.com:goalgorilla/gdpr_consent into feature/DS-5171-data-policy-revisions

ohorbatiuk committed:
DS-5171 by chmez: Delete deprecated code.

ohorbatiuk committed:
DS-5171 by chmez:

1. Add description to help page of the module.

2. Hide user consents overview page in the administrative menu.

ohorbatiuk committed:
DS-5171 by chmez: Show only a few first words of revision log message on revisions overview page.

ohorbatiuk committed:
DS-5171 by chmez: Add description to form for reverting and create method for checking access to reverting.

ohorbatiuk committed:
DS-5171 by chmez: Add a wrapper for data policy content.

ohorbatiuk committed:
DS-5171 by chmez: Improve pages controller.

bramtenhove commented on src/DataPolicyAccessControlHandler.php:
```diff
@@ -0,0 +1,37 @@
+   */
+  public function access(EntityInterface $entity, $operation, AccountInterface $account = NULL, $return_as_object = FALSE) {
+    $access = parent::access($entity, $operation, $account, $return_as_object);
+    $x = 1;
```
@OH17L this seems out of place. Perhaps remove it?

OH17L replied:
@bramtenhove, yes, I will delete it.

bramtenhove commented on src/DataPolicyTranslationHandler.php:
```diff
@@ -0,0 +1,12 @@
+/**
+ * Defines the translation handler for data_policy.
+ */
+class DataPolicyTranslationHandler extends ContentTranslationHandler {
```
@OH17L this class does nothing. Shouldn't we just point to the normal class? E.g. `ContentTranslationHandler`?

OH17L replied:
@bramtenhove, I will check it.

bramtenhove commented on src/Entity/DataPolicy.php:
```diff
@@ -16,41 +17,95 @@
+ *     "storage" = "Drupal\gdpr_consent\DataPolicyStorage",
+ *     "view_builder" = "Drupal\Core\Entity\EntityViewBuilder",
+ *     "views_data" = "Drupal\gdpr_consent\Entity\DataPolicyViewsData",
+ *     "translation" = "Drupal\gdpr_consent\DataPolicyTranslationHandler",
```
Perhaps use `Drupal\content_translation\ContentTranslationHandler` as the `DataPolicyTranslationHandler` is empty?

OH17L replied:
@bramtenhove, I will check it.

bramtenhove replied:
I've been testing translations. After removing this I got errors when enabling content translation for Data Policy.
Maybe put this back?

bramtenhove commented on src/Form/DataPolicySettingsForm.php:
```diff
@@ -0,0 +1,53 @@
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
```
@OH17L, just wondering. But could we not simply use `{@inheritdoc}` here (and also below)?

OH17L replied:
@bramtenhove you are right. I will change it.

bramtenhove commented:
@OH17L I noticed that when I create a new revision it also copies the revision comment I wrote in the previous one.
I think we should not autofill the revision comment from the previous revision. So just leave it empty.

ohorbatiuk committed:
DS-5171 by chmez: Delete unneeded code.

ohorbatiuk committed:
DS-5171 by chmez: Make "Revision log message" field is empty when creating a new revision.

bramtenhove commented:
@OH17L, can you explain how I can translate the data policy? When I reverted back to the previous commit (with the custom translation handler class) I am able to create a data policy in a specific language. However I cannot find a button or anything to translate the data policy. Any clue?

ohorbatiuk committed:
DS-5171 by chmez:

1. Delete canonical pages of the Data policy entity type.

2. Use file-template instead of inline template for revision teasers on the overview page.

ohorbatiuk committed:
DS-5171 by chmez: Add a field for choice language on a form for creating a new revision.

bramtenhove commented:
@OH17L, I will merge this PR. We can further improve translations in a different story.
